include/roaring/containers/containers.h

```diff
@@ -55,12 +55,25 @@ extern "C" { namespace roaring { namespace internal {
  * A shared container is a wrapper around a container
  * with reference counting.
  */
-
+#if CROARING_C_ATOMIC
+STRUCT_CONTAINER(shared_container_s) {
+    container_t *container;
+    uint8_t typecode;
+    _Atomic(uint32_t) counter;  // to be managed atomically
```

Comment:
Are you game do a PR on this PR?

I am quite open to changing this code, but it would be more efficient if you did your own changes. Then we could discuss them.

I do not pretend that I did things perfectly or even well.

Reply:
I could give it a try, won't be till at least tomorrow evening

Reply:
It is ok. Something is going on with Win32 and I can't figure out what.

Reply:
Note that I have resolved the Windows issues.